### ISSUE TITLE
C3.L2.S4 vs C2 re:L5-L7): Clarification on Core 2 Requirements

### ISSUE THREAD

**Issue by SherylHohman** [udacity/dlnd-issue-reports#498]
Core 2 indicates that Lessons after Project 1 are optional.
"You have me all requirements for Core 2. Continue to Core 3" (paraphrased)
This is message appears at the top of the Core 2 page which lists al the Lessons in Core 2.

Andrew Trask's lesson, and Keras and Tensorflow are all listed *after* Project 1.
And even before I looked at any of them the message "You have completed all Part 2 requirements!
PROCEED TO PART 3" with a button and link to Core 3.

Also on the Top Level Page for the course, the Left-hand Core List shows Core 2 at 100% complete - even though I am 50% through Andrew Trasks lesson, and have not started Keras or Tensorflow. 

https://classroom.udacity.com/nanodegrees/nd101/parts/73a7fe8e-4354-4362-939d-a8bf2bae870d

-----
However, Core3, Lesson2, Section4 video:

https://classroom.udacity.com/nanodegrees/nd101/parts/b9c4c3c3-b524-427b-8832-9d0748f14a2e/modules/cb574ac4-7144-4ba5-97b9-1c1265525ff8/lessons/37492b45-ee34-4dc3-a2e8-076149f92562/concepts/87712da9-5e20-4d75-b067-aa5f354e4a85

references Keras code that we were supposed to be familiar with.

That must have been in the "Optional?" Keras modlue from Core2.

-------
Recommendation: If Keras, TensorFlow, and/or Sentiment Analysis are pre-requisites to Core 3, please fix the "Course Progress" settings and messages.
If not, please update the video, or add an appropriate comment in the notes below the video, indicating what we need to know, or indicating that we can ignore such references.

**Comment by lcrucks** [comment]
Hi Sheryl,
'Optional' here means that it isn't *strictly* required to finish the course.  For example, if there were no homework grades in a class, and everything was based on tests, the homework is, strictly speaking, 'optional', even in most people will fail the tests without doing the homework.   In the Deep Learning Nanodegree, the only thing that is required to earn your certificate is to complete the projects, so you have indeed finished all 'requirements' in that section.

I do understand this language is confusing; I can forward this to engineering so they can consider site-wide changes to the wording of these messages.

For your own progress, it's best to consider  that most things you learn will be built on later, so they're not really optional in the sense of learning the material, but you aren't submitting anything to be reviewed, so it's optional in that sense.

**Comment by SherylHohman** [comment]
The problem with calling this "optional" is that it is impossible to work through the next section if the so-called optional parts were skipped.  The lecture and materials refer to that work.  It in truth is a **pre-requisite** for the next module.

**Comment by SherylHohman** [comment]
By your line of thinking then you might as well label the entire body of this course as "optional", which is pointless.  Being that the progress meters exclude *that* material, but none of the remaining course material is a statement on how this is being treated very differently in metrics and UI than it is for the lectures.
